htdocs/compta/tva/index.php

```diff
@@ -43,61 +44,89 @@
+		} elseif ($month_current >= 10 && $month_current <= 12) {
+			$month_start = 7;
+			$month_end = 9;
+		} elseif ($month_current >= 1 && $month_current <= 3) {
+			$month_start = 10;
+			$month_end = 12;
+			$year--;
 		} else {
-			if (empty($conf->global->MAIN_INFO_VAT_RETURN) || $conf->global->MAIN_INFO_VAT_RETURN == 2) { // quaterly vat, we take last past complete quarter
-				$date_start = dol_time_plus_duree(dol_get_first_day($year_start, $current_date['mon'], false), -3 - (($current_date['mon'] - $conf->global->SOCIETE_FISCAL_MONTH_START) % 3), 'm');
-				$date_end = dol_time_plus_duree($date_start, 3, 'm') - 1;
-			} elseif ($conf->global->MAIN_INFO_VAT_RETURN == 3) { // yearly vat
-				if ($current_date['mon'] < $conf->global->SOCIETE_FISCAL_MONTH_START) {
-					if (($conf->global->SOCIETE_FISCAL_MONTH_START - $current_date['mon']) > 6) {	// If period started from less than 6 years, we show past year
-						$year_start--;
-					}
-				} else {
-					if (($current_date['mon'] - $conf->global->SOCIETE_FISCAL_MONTH_START) < 6) {	// If perdio started from less than 6 years, we show past year
-						$year_start--;
-					}
-				}
-				$date_start = dol_get_first_day($year_start, $conf->global->SOCIETE_FISCAL_MONTH_START, false);
-				$date_end = dol_time_plus_duree($date_start, 1, 'y') - 1;
-			} elseif ($conf->global->MAIN_INFO_VAT_RETURN == 1) {	// monthly vat, we take last past complete month
-				$date_start = dol_time_plus_duree(dol_get_first_day($year_start, $current_date['mon'], false), -1, 'm');
-				$date_end = dol_time_plus_duree($date_start, 1, 'm') - 1;
-			}
+			$month_start = 1;
+			$month_end = 3;
 		}
+		$date_start = dol_get_first_day($year, $month_start);
+		$date_end = dol_get_last_day($year, $month_end);
+	} elseif ($conf->global->MAIN_INFO_VAT_RETURN == 3) {
+		// annual
+		$date_start = dol_get_first_day($year_current, 1);
+		$date_end = dol_get_last_day($year_current, 12);
 	} else {
-		if ($q == 1) {
-			$date_start = dol_get_first_day($year_start, 1, false);
-			$date_end = dol_get_last_day($year_start, 3, false);
-		}
-		if ($q == 2) {
-			$date_start = dol_get_first_day($year_start, 4, false);
-			$date_end = dol_get_last_day($year_start, 6, false);
+		// monthly by default
+		$year = $year_current;
+		$month_last = $month_current - 1;
+		if ($month_last <= 0) {
+			$month_last = $month_last + 12;
+			$year--;
 		}
-		if ($q == 3) {
-			$date_start = dol_get_first_day($year_start, 7, false);
-			$date_end = dol_get_last_day($year_start, 9, false);
-		}
-		if ($q == 4) {
-			$date_start = dol_get_first_day($year_start, 10, false);
-			$date_end = dol_get_last_day($year_start, 12, false);
+		$date_start = dol_get_first_day($year, $month_last);
+		$date_end = dol_get_last_day($year, $month_last);
+	}
+} else {
+	// Date range
+	$year = GETPOST("year", "int");
+	if (empty($year)) {
+		$year_current = dol_print_date("%Y", dol_now());
+		if ($conf->global->SOCIETE_FISCAL_MONTH_START > date('m')) $year_current--;
```

Comment:
Same "date" function should not be used to get month, use instead 
dol_print_date(dol_now(), "%m")